File: 880/cantor.c
#include <stdio.h>
#include <math.h>

double get_n(long s)
{
	double y = sqrt(2*s + 0.25);

	return y - 0.5;
}

long get_sum(long n)
{
	return (n * (n + 1)) / 2;
}

int main(int argc, char *argv[])
{
	long t;
	double y;
	long x, z;
	long target, sum_floor, term_no;
	long denom, numer, i;

	while(scanf("%ld", &t) == 1)
	{
		y = get_n(t);

		x = (long)y;

		if(y - x < 0.0000001)
			x = x - 1;

		z = (long)ceil(y);

		target = z + 1;

		sum_floor = get_sum(x);

		term_no = t - sum_floor;

		denom = 0;
		for(i=0; i<term_no; i++)
			denom++;

		numer = target - denom;

		printf("%ld/%ld\n", numer, denom);
	}

	return 0;
}

File: 880/cantor.py
import sys
import math

# Time limit exceeded 

def get_n(s):
	# y = n + 0.5
	y = math.sqrt(2*s + 0.25)

	return (y - 0.5)

def get_sum(n):
	return (n * (n + 1)) // 2


for line in sys.stdin:
	t = int(line)

	y = get_n(t)
	x = math.floor(y)

	if x == y:
		x = x - 1

	z = math.ceil(y)

	target = int(z + 1)

	s_floor = get_sum(x)

	term_no = t - s_floor

	denom = 0
	for i in range(term_no):
		denom = denom + 1

	numer = target - denom

	print("%d/%d"%(numer, denom))
